zcash_primitives/src/transaction/components/orchard/burn_validation.rs

```diff
@@ -0,0 +1,118 @@
+use std::fmt;
+
+use orchard::note::AssetBase;
+
+use super::Amount;
+
+// FIXME: Consider making tuple (AssetBase, Amount) a new type.
+
+#[derive(Debug)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+pub enum BurnError {
+    DuplicateAsset,
+    NativeAsset,
+    ZeroAmount,
+}
+
+impl fmt::Display for BurnError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            BurnError::DuplicateAsset => write!(f, "Encountered a duplicate asset to burn."),
+            BurnError::NativeAsset => write!(f, "Cannot burn a native asset."),
+            BurnError::ZeroAmount => write!(f, "Cannot burn an asset with a zero amount."),
+        }
+    }
+}
+
+/// Validates burns for a bundle by ensuring each asset is unique, non-native, and has a non-zero value.
+///
+/// Each burn element is represented as a tuple of `AssetBase` and `Amount`, where `AssetBase` identifies
+/// the asset to be burned and `Amount` is the quantity to burn.
+///
+/// # Arguments
+///
+/// * `burns` - A vector of burns, where each burn is represented as a tuple of `AssetBase` and `Amount`.
+///
+/// # Errors
+///
+/// Returns a `BurnError` if:
+/// * Any asset in the list of burns is not unique (`BurnError::DuplicateAsset`).
+/// * Any asset in the list of burns is native (`BurnError::NativeAsset`).
+/// * Any asset in the list of burns has a zero amount (`BurnError::ZeroAmount`).
+pub fn validate_bundle_burn(bundle_burn: &Vec<(AssetBase, Amount)>) -> Result<(), BurnError> {
+    let mut asset_set = std::collections::HashSet::<AssetBase>::new();
+
+    for (asset, amount) in bundle_burn {
+        if !asset_set.insert(*asset) {
+            return Err(BurnError::DuplicateAsset);
+        }
+        if asset.is_native().into() {
+            return Err(BurnError::NativeAsset);
+        }
+        // FIXME: check for negative amounts?
+        if i64::from(amount) == 0 {
+            return Err(BurnError::ZeroAmount);
+        }
```

Comment:
Yes, and change the error to `NonPossitiveAmount`

Reply:
Done.